src/azul/indexer/document.py

```diff
@@ -210,6 +210,7 @@ def __str__(self) -> str:
 class FieldType(Generic[N, T], metaclass=ABCMeta):
     shadowed: bool = False
     es_sort_mode: str = 'min'
+    es_type: Optional[str] = None
```

Comment:
All concrete subclasses supply a value for this so it doesn't need to be optional.